core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -1044,7 +1045,7 @@ public static void cleanSegments(CarbonTable table,
           Long fileTimestamp = CarbonUpdateUtil.getTimeStampAsLong(indexFile
               .substring(indexFile.lastIndexOf(CarbonCommonConstants.HYPHEN) + 1,
                   indexFile.length() - CarbonTablePath.INDEX_FILE_EXT.length()));
-          if (CarbonUpdateUtil.isMaxQueryTimeoutExceeded(fileTimestamp) || forceDelete) {
+          if (TrashUtil.isTrashDataTimeout(fileTimestamp) || forceDelete) {
```

Comment:
Also update this behavior change in document, now `max.query.execution.time`  expire will not delete the data but trash time has to expire also.

Reply:
yes, the requirement is just like it.